x-pack/platform/plugins/shared/dataset_quality/public/components/dataset_quality_details/overview/index.tsx

```diff
@@ -39,6 +44,23 @@ export function Overview() {
       {isNonAggregatable && <AggregationNotSupported dataStream={dataStream} />}
       <OverviewHeader handleRefresh={handleRefresh} />
       <EuiSpacer size="m" />
+      {!hasFailureStore && canUserReadFailureStore && (
+        <div style={{ marginBottom: 16 }}>
+          <EuiCallOut
+            color="warning"
+            iconType="warning"
+            title={i18n.translate('xpack.datasetQuality.noFailureStoreTitle', {
+              defaultMessage: 'Failure store is not enabled for this data stream.',
+            })}
+          >
+            <EuiLink href={FAILURE_STORE_DOCS_URL} external>
+              {i18n.translate('xpack.datasetQuality.learnMore', {
+                defaultMessage: 'Learn how to enable it',
+              })}
+            </EuiLink>
+          </EuiCallOut>
+        </div>
+      )}
       {!dataStreamSettingsLoading && !canUserReadFailureStore && (
```

Comment:
This is what I was proposing in my comment, to only show if `settings` has finished